dask/dataframe/io.py

```diff
@@ -709,12 +710,23 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
         name = 'read-hdf-' + token
 
         dsk = dict(((name, i), (_pd_read_hdf, path, key, lock,
-                                 {'start': s,
-                                  'stop': s + chunksize,
-                                  'columns': empty.columns}))
-                    for i, s in enumerate(range(start, stop, chunksize)))
-
-        divisions = [None] * (len(dsk) + 1)
+                                {'start': s,
+                                 'stop': s + chunksize,
+                                 'columns': empty.columns}))
+                   for i, s in enumerate(range(start, stop, chunksize)))
+
+        if index:
+            divisions = list()
+            for i in range(start, stop, chunksize):
+                kwargs = {'start': i, 'stop': i + 1}
+                a = _pd_read_hdf(path, key, lock, kwargs)
+                divisions.append(a.index[0])
```

Comment:
Or perhaps we ask the user to make this guarantee?

Also, https://github.com/dask/dask/issues/955